guac_core/src/storages/in_memory.rs

```diff
@@ -0,0 +1,100 @@
+use channel_client::types::{Channel, ChannelStatus};
+use clarity::Address;
+use crypto::CryptoService;
+use failure::Error;
+use futures::future::ok;
+use futures::Future;
+use num256::Uint256;
+use payment_contract::ChannelId;
+use qutex::Qutex;
+use rand;
+use rand::prelude::*;
+use rand::RngCore;
+use std::collections::HashMap;
+use storage::Storage;
+use CRYPTO;
+
+struct ChannelData {
+    url: String,
+
+    // TODO: Channel structure needs some changes but we can reuse it at this point
+    channel: Channel,
+}
+
+/// A in-memory storage that stores data in
+pub struct InMemoryStorage {
+    channels: Qutex<HashMap<ChannelId, ChannelData>>,
+}
+
+impl InMemoryStorage {
+    pub fn new() -> Self {
+        Self {
+            channels: Qutex::new(HashMap::new()),
+        }
+    }
+}
+
+impl Storage for InMemoryStorage {
+    /// Registers new counterparty
+    ///
+    /// * `url` - Remote URL
+    /// * `url` - Remote ETH address
+    /// * `balance` - Our initial deposit
+    fn register(
+        &self,
+        url: String,
+        address: Address,
```

Comment:
Be explicit about *whose* (counterparty in this case) address you're dealing with. Issues stemming for bad naming are trivial to prevent.

Reply:
Addressed already in recent commits. Now its "address0, address1, balance0, balance1" where first in pair is "source" (us) and second in pair is "destination" (them).